### PR TITLE
fix(client): Visible success/error messages stay visible on view render

### PR DESCRIPTION
**Pull request by shane-tomlinson**
Separate `enableSubmitIfValid` into two functions, `onFormChange`
and `enableSubmitIfValid`. `onFormChange` takes care of hiding
status messages `enableSubmitIfValid` only enables the submit button
if the form is valid. `enableSubmitIfValid` is called after
the view is rendered.

Because of this change, the `enableSubmitIfValid` override in
`settings/communication_preferences` is no longer needed.

A new test template was needed to prevent other tests from failing.

This is an extraction from https://github.com/mozilla/fxa-content-server/pull/4370

@vladikoff or @philbooth - r?

### CODE REVIEW

File: app/tests/spec/views/form.js [hunk 1]
```diff
@@ -101,10 +100,7 @@ define(function (require, exports, module) {
         model: model
       });
 
-      return view.render()
-        .then(function () {
-          $('#container').html(view.el);
```
Comment (shane-tomlinson) [inline app/tests/spec/views/form.js]: Only writing to the DOM when necessary speeds up this suite!

File: app/tests/spec/views/form.js [hunk 2]
```diff
@@ -605,7 +587,7 @@ define(function (require, exports, module) {
             } catch (e) {
               defer.reject(e);
             }
-          }, 500);
+          }, 20);
```
Comment (philbooth) [inline app/tests/spec/views/form.js]: ...and not delaying for half a second also has an effect I imagine!

**Comment by philbooth** [comment]
LGTM, r+! 👍

**Comment by shane-tomlinson** [comment]
@philbooth - mind doing a review of the 2nd commit on this? I made some substantial changes to ensure `isSuccessVisible` and `isErrorVisible` return the correct value if the status messages are displayed via the template.

**Comment by philbooth** [comment]
Still looking good to me @shane-tomlinson! 👍

**Pull request merged**